### DISCUSSION TITLE
UUID for Vector_ids Version 1.1

### DISCUSSION THREAD

**Discussion by STAROFWIND** [milvus-io/milvus#14171] [Q&A and General discussion] [answered]
Hi all, 
I'm working on milvus v1.1. And I want to use uuid (python) or other type like: string,...  for vector ids. Is It possible ? If yes, could you please give me sơm hints for this. Thank you so much.

**Comment by yhmo** [comment, marked as answer]
Milvus v1.x only support int64 as uid. Some user use other database(postgres, mysql) to store mapping relationship between int64 uid and string uid. Do ANN search by milvus, milvus return topk result with int64 id, then use the int64 id to find string id from the other database(postgres, mysql).
Currently milvus v2.0 also only siupport int64 ad uid. We plan to support string type field in v2.1

**Comment by STAROFWIND** [comment]
Thank you for your reply. 
For one who want to use uuid (128 bit). You can accept a small collision when you hash to 64bit. Using https://stackoverflow.com/questions/3530294/how-to-generate-unique-64-bits-integers-from-python can help you. Best Regard !!!
